src/Logging/Logging.Testing/src/CollectDumpAttribute.cs

```diff
@@ -12,7 +17,23 @@ namespace Microsoft.Extensions.Logging.Testing
     /// This currently only works in Windows environments
     /// </remarks>
     [AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]
-    public class CollectDumpAttribute : Attribute
+    public class CollectDumpAttribute : Attribute, ITestMethodLifecycle
```

Comment:
This is probably a good place to start reviewing as any. This is an example of the kind of thing that's possible now.